### PR TITLE
Add a debug flag that enables wire logging for git http requests

### PR DESCRIPTION
**Pull request by cjneasbi**

### CODE REVIEW

File: dockerfile-image-update/pom.xml [hunk 1]
```diff
@@ -144,6 +144,16 @@
...
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+            <version>4.9.1</version>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>logging-interceptor</artifactId>
+            <version>4.9.1</version>
+        </dependency>
```
Comment (cjneasbi) [inline dockerfile-image-update/pom.xml:156]: For instrumenting the client with logging functionality.

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/CommandLine.java [hunk 2]
```diff
@@ -181,27 +194,65 @@ public static Class<?> loadCommand(Set<ClassPath.ClassInfo> allClasses, String c
...
+    }
+
+    public static String gitApiToken(final Function<String, String> envFunc, final Consumer<Integer> exitFunc) {
+      final String token = envFunc.apply("git_api_token");
+      if (Objects.isNull(token)) {
+          log.error("Please provide GitHub token in environment variables.");
+          exitFunc.accept(3);
+      }
+      return token;
+    }
```
Comment (cjneasbi) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/CommandLine.java:208]: Moved from within initializeDockerfileGithubUtil

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/CommandLine.java [hunk 3]
```diff
@@ -181,27 +194,65 @@ public static Class<?> loadCommand(Set<ClassPath.ClassInfo> allClasses, String c
...
+      return gitApiUrl(ns, System::getenv);
+    }
+
+    public static String gitApiUrl(final Namespace ns, final Function<String, String> envFunc) throws IOException {
+      return Optional.ofNullable(
+                 Optional.ofNullable(ns.getString(GIT_API))
+                   .orElse(envFunc.apply("git_api_url"))
+             )
+             .orElseThrow(() -> new IOException("No Git API URL in environment variables nor on the commmand line."));
+    }
```
Comment (cjneasbi) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/CommandLine.java:220]: Moved from within initializeDockerfileGithubUtil

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/Parent.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public void execute(final Namespace ns, DockerfileGitHubUtil dockerfileGitHubUti
             log.error("Could not initialize the Image tage store. Exception: ", e);
         }
 
-        if (ns.get(Constants.SKIP_PR_CREATION)) {
+        if (ns.getBoolean(Constants.SKIP_PR_CREATION)) {
```
Comment (cjneasbi) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/Parent.java:57]: Fixes error of trying to cast Object to boolean

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 5]
```diff
@@ -39,7 +39,7 @@ public DockerfileGitHubUtil(GitHubUtil gitHubUtil) {
         this.gitHubUtil = gitHubUtil;
     }
 
-    protected GitHubUtil getGitHubUtil() { return gitHubUtil; }
+    public GitHubUtil getGitHubUtil() { return gitHubUtil; }
```
Comment (cjneasbi) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java:42]: Visible for testing

**Review by afalko — APPROVED** [review]

**Pull request merged**